File: Python/Program/Car.py
'''
Saya Nabil Hanif Achmaddiredja mengerjakan Latihan Praktikum 4 dalam 
mata kuliah Desain dan Pemrograman Berorientasi Objek untuk 
keberkahanNya maka saya tidak melakukan kecurangan seperti yang telah dispesifikasikan. Aamiin.
'''

from Vehicle import Vehicle

#Private Attributes
class Car(Vehicle):
    #Private Attributes
    __jumlahKursi = 0
    __jumlahPintu = 0
    
    # Constructor
    def __init__(self, platNomor="", merk="", tahunProduksi="", warna="", kapasitasTangki="", jumlahKursi=0, jumlahPintu=0):
        super().__init__(platNomor, merk, tahunProduksi, warna, kapasitasTangki)
        self.__jumlahKursi = jumlahKursi
        self.__jumlahPintu = jumlahPintu
    
    # Getter and Setter
    # Getter 
    def getJumlahKursi(self):
        return self.__jumlahKursi

    def getJumlahPintu(self):
        return self.__jumlahPintu
    
    # Setter 
    def setJumlahKursi(self, jumlahKursi):
        self.__jumlahKursi = jumlahKursi

    def setJumlahPintu(self, jumlahPintu):
        self.__jumlahPintu = jumlahPintu

File: Cpp/Program/Car.cpp
/*Saya Nabil Hanif Achmaddiredja mengerjakan Latihan Praktikum 4 dalam 
mata kuliah Desain dan Pemrograman Berorientasi Objek untuk 
keberkahanNya maka saya tidak melakukan kecurangan seperti yang telah dispesifikasikan. Aamiin.*/


#include <iostream>
#include <string>
#include "Vehicle.cpp"

using namespace std;

class Car : public Vehicle
{
    private:
        int jumlahKursi;
        int jumlahPintu;

    public:
        /*Constructor*/
        Car(){

        }

        Car(string platNomor, string merk, string tahunProduksi, string warna, string kapasitasTangki, int jumlahKursi, int jumlahPintu) : Vehicle(platNomor, merk, tahunProduksi, warna, kapasitasTangki) {
            this->jumlahKursi = jumlahKursi;
            this->jumlahPintu = jumlahPintu;
        }

        /*Getter & Setter*/
        //Getter
        int getJumlahKursi() {
            return this->jumlahKursi;
        }

        int getJumlahPintu() {
            return this->jumlahPintu;
        }

        // Setter methods
        void setJumlahKursi(int jumlahKursi) {
            this->jumlahKursi = jumlahKursi;
        }

        void setJumlahPintu(int jumlahPintu) {
            this->jumlahPintu = jumlahPintu;
        }

    ~Car()
    {}
};